python_lib/qss_compiler/mlir/dialects/_pulse_ops_ext.py

```diff
@@ -0,0 +1,138 @@
+# (C) Copyright IBM 2024.
+#
+# This code is part of Qiskit.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    from ..ir import (
+        FlatSymbolRefAttr,
+        StringAttr,
+        TypeAttr,
+        FunctionType,
+        ArrayAttr,
+    )
+    from ..ir import *  # noqa: F401, F403
+    from .pulse import *  # noqa: F401, F403
+    from .quir import (  # noqa: F401, F403
+        AngleType,
+        ConstantOp as QUIRConstantOp,
+    )
+    from .._mlir_libs._ibmDialectsPulse import *  # noqa: F401, F403
+    from ._ods_common import (  # noqa: F401
+        get_default_loc_context as _get_default_loc_context,
+    )
+except ImportError as e:
+    raise RuntimeError("Error loading imports from extension module") from e
+
+from typing import Any, List, Optional, Union  # noqa: F401
+
+
+class CallSequenceOp:
+    def __init__(self, res, callee, operands_, *, loc=None, ip=None):
+        super().__init__(res, FlatSymbolRefAttr.get(str(callee)), operands_)
+
+
+class Frame_CreateOp:
+    """Specialization for the Frame_Create op class."""
+
+    def __init__(self, uid, *, amp=None, freq=None, phase=None, loc=None, ip=None):
+        super().__init__(
+            StringAttr.get(str(uid)),
+            initial_amp=amp,
+            initial_freq=freq,
+            initial_phase=phase,
+            loc=loc,
+            ip=ip,
+        )
+
+
+class MixFrameOp:
+    """Specialization for the MixFrame op class."""
+
+    def __init__(
+        self,
+        port,
+        frame,
+        *,
+        initial_amp=None,
+        initial_freq=None,
+        initial_phase=None,
+        loc=None,
+        ip=None
+    ):
+        super().__init__(
+            port,
+            StringAttr.get(str(frame)),
+            initial_amp=initial_amp,
+            initial_freq=initial_freq,
+            initial_phase=initial_phase,
+            loc=loc,
+            ip=ip,
+        )
+
+
+class PlayOp:
+    def __init__(self, target, wfr, *, loc=None, ip=None):
+        angle, ssb, amplitude = None, None, None
```

Comment:
Will this override an angle, sub, or amplitude passed as a keyword argument? Do we want to prevent setting these values?

Reply:
Updated the def in https://github.com/openqasm/qe-compiler/pull/276/commits/2171b6be9126158f37c70ad2300798a7a35b9da7.